### PR TITLE
[#319] Fix null versions

### LINKED ISSUE #319: Support "null" version deletion
We should support `null` version deletion when versioning is enabled 

Check the following tests:
* `test_s3.test_versioning_obj_plain_null_version_removal`
* `test_s3.test_versioning_obj_plain_null_version_overwrite`

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #319 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: api/data/info.go [hunk 1]
```diff
@@ -68,6 +68,15 @@ func (b *BucketInfo) CORSObjectName() string { return bktCORSConfigurationObject
 // Version returns object version from ObjectInfo.
 func (o *ObjectInfo) Version() string { return o.ID.String() }
 
+// NullableVersion returns object version from ObjectInfo.
+// Return "null" if "S3-Versions-unversioned" header present.
+func (o *ObjectInfo) NullableVersion() string {
```
Comment (masterSplinter01) [inline api/data/info.go:73]: I think, the `Nullable` prefix seems redundant a bit. 

As I see `Version` function is used only in `NullableVersion` and `TagsObject`. Maybe it'd be more clear to write something like this: 
```
// Version returns object version from ObjectInfo.
// Return "null" if "S3-Versions-unversioned" header present.
func (o *ObjectInfo) Version() string {
	if _, ok := o.Headers["S3-Versions-unversioned"]; ok {
		return "null"
	}
	return o.o.ID.String()
}
...
func (o *ObjectInfo) TagsObject() string { return ".tagset." + o.Name + "." + o.ID.String() }
```

Comment (KirillovDenis) [reply api/data/info.go:73]: In most cases we want to treat version as object id (for storing in headers and so on). `Version()` function is used not only in `NullableVesion` and `TagsObject` (we use it outside `data` package). But currently `NullableVersion` is used only once (to encode `list-object-versions` response) so we can define similar function there and drop it from this package

Comment (alexvanin) [reply api/data/info.go:73]: I think it is okay to keep it there, though. You may want to specify in the comment, that `NullableVersion` is required for listing object version.

**Review by alexvanin — APPROVED** [review]

File: docs/s3_test_results.md [hunk 2]
```diff
@@ -383,17 +383,17 @@ Compatibility: 9/6/8 out of 11
 
 ## Versioning
 
-Compatibility: 16/19/24 out of 26
```
Comment (masterSplinter01) [inline docs/s3_test_results.md:386]: As I see this PR fixes the test `test_versioning_bucket_multipart_upload_return_version_id` and deletion of buckets (#320) too, maybe we should update test result here?

Comment (KirillovDenis) [reply docs/s3_test_results.md:386]: This test was fixed early https://github.com/nspcc-dev/neofs-s3-gw/pull/321

Comment (masterSplinter01) [reply docs/s3_test_results.md:386]: Okay, but anyway the PR closes #320 too.

**Pull request merged**